### PR TITLE
Transformer Encoder - Representation Parameter Fix

### PR DESCRIPTION
**Pull request by connor-mccorm**
This PR addresses this training failure:
```
Exception: Embedding representation None not supported. (type: RayTaskError, retryable: true)
```

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -27,12 +27,15 @@
...
     INPUT_FEATURES,
+    MODEL_ECD,
+    MODEL_GBM,
     NAME,
     OUTPUT_FEATURES,
     TRAINER,
     TYPE,
 )
 from ludwig.hyperopt.run import hyperopt
+from ludwig.schema import get_schema
```
Comment (justinxzhao) [inline tests/conftest.py:38]: Can you import `from ludwig.config_validation.validations import get_schema`? Want to try to remove `ludwig.schema.__init__.py`

File: tests/ludwig/schema/test_schema_fields.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+from ludwig.constants import ALLOF, ENCODER, ENUM, INPUT_FEATURES, ITEMS, PROPERTIES, THEN
```
Comment (justinxzhao) [inline tests/ludwig/schema/test_schema_fields.py:1]: If we keep these tests, add them tests to `tests/ludwig/config_validation/`

File: tests/ludwig/schema/test_schema_fields.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+def parse_schema(input_dict: dict, key_list: list):
+    """Parse a nested dictionary by a list of keys."""
+    parsed_dict = input_dict
+    for key in key_list:
+        parsed_dict = parsed_dict[key]
+    return parsed_dict
+
+
+def test_transformer_encoder_representations(ecd_schema):
```
Comment (justinxzhao) [inline tests/ludwig/schema/test_schema_fields.py:12]: I'm not convinced that we absolutely need this test, since it seems like it's testing implementation details, and we already have tests for `StringOptions`.

Guidance on not testing implementation details:
https://testing.googleblog.com/2013/08/testing-on-toilet-test-behavior-not.html

Comment (connor-mccorm) [reply tests/ludwig/schema/test_schema_fields.py:12]: Ya I feel like it's a little pedantic, however Travis asked specifically for this test [here](https://docs.google.com/spreadsheets/d/1By3vMX17dVwpvs0lyiLim5su4gp7pHlA3L7542WtbaI/edit?disco=AAAApOfcHF4)
```
Travis Addair
12:36 PM Yesterday
And add unit test (enumerate all possible values, check does not allow None).
```
So I interpreted that as I should build this test. Happy to remove though if we feel it's unnecessary.

File: tests/ludwig/schema/test_schema_fields.py (hunk 2 above)
Comment (connor-mccorm) [reply tests/ludwig/schema/test_schema_fields.py:1]: Done!

File: tests/conftest.py (hunk 1 above)
Comment (connor-mccorm) [reply tests/conftest.py:38]: Done!

File: tests/ludwig/schema/test_schema_fields.py (hunk 3 above)
Comment (justinxzhao) [reply tests/ludwig/schema/test_schema_fields.py]: It looks like there's already a test for `allow_none=False` for `StringOptions`.

https://github.com/ludwig-ai/ludwig/blob/master/tests/ludwig/marshmallow/test_fields_misc.py#L20-L34

I'd say we can do without this test.

Comment (connor-mccorm) [reply tests/ludwig/schema/test_schema_fields.py]: Sounds good, will remove!

**Review by justinxzhao — APPROVED** [review]
